### PR TITLE
etcd(ticdc): not skip etcd worker tick if nothing is changed

### LINKED ISSUE #7357: not skip etcd worker tick if nothing is changed
https://github.com/pingcap/tiflow/blob/c5b1dd0edcf2d1b03838619fdb27acab908b1f27/pkg/orchestrator/reactor_state.go#L262-L282

always add a pending patch, but nothing is changed,  that will make etcd worker skip one tick

### PR DESCRIPTION
**Pull request by sdojjy**
### What problem does this PR solve?

Issue Number: close #7357

### What is changed and how it works?
https://github.com/pingcap/tiflow/blob/c5b1dd0edcf2d1b03838619fdb27acab908b1f27/pkg/orchestrator/reactor_state.go#L262-L282
always add a pending patch even if nothing is changed
in this PR, 
1. check if something is really committed to etcd server, if not, continue the current tick loop.
2. try to commit changes at the end of tick 

#### Tests 

 - Unit test
 - Integration test

### Release note 

```release-note
`None`.
```

### CODE REVIEW

**Review by overvenus — APPROVED** [review]
Could you add a test?

File: pkg/orchestrator/etcd_worker.go [hunk 1]
```diff
@@ -163,9 +163,11 @@ func (worker *EtcdWorker) Run(ctx context.Context, session *concurrency.Session,
 	}
 
 	var (
-		pendingPatches [][]DataPatch
-		exiting        bool
-		sessionDone    <-chan struct{}
+		pendingPatches  [][]DataPatch
+		commitedChanges int
```
Comment (3AceShowHand) [inline pkg/orchestrator/etcd_worker.go:167]:
```suggestion
		committedChanges int
```

Comment (sdojjy) [reply pkg/orchestrator/etcd_worker.go]: fixed

File: pkg/orchestrator/etcd_worker.go [hunk 2]
```diff
@@ -229,58 +231,75 @@ func (worker *EtcdWorker) Run(ctx context.Context, session *concurrency.Session,
...
 			}
+			return false, nil
+		}
+		if retry, err = tryCommitPendingPatches(); err != nil {
+			return err
+		}
+		if retry {
+			continue
+		}
+		if exiting {
```
Comment (3AceShowHand) [inline pkg/orchestrator/etcd_worker.go:258]: it looks, we should check `exsting` before `retry`

Comment (sdojjy) [reply pkg/orchestrator/etcd_worker.go:258]: done

**Review by asddongmen — APPROVED** [review]

**Pull request merged**